Tweak makefiles to use /usr as default prefix.

--- a/Makefile.am
+++ b/Makefile.am
@@ -39,7 +39,7 @@
 CFLAGS += -Wno-pointer-sign
 endif
 else
-CPPFLAGS += -I. -I- -I/usr/local/include
+CPPFLAGS += -I. -I- -I/usr/include
 endif
 endif
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -54,7 +54,7 @@
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wundef -Wbad-function-cast \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Winline -Wcast-align
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@am__append_4 = -DDEBUG
-@IS_GCC4_TRUE@@IS_GCC_TRUE@am__append_5 = -iquote. -I/usr/local/include
+@IS_GCC4_TRUE@@IS_GCC_TRUE@am__append_5 = -iquote. -I/usr/include
 @IS_DEBUG_TRUE@@IS_GCC4_TRUE@@IS_GCC_TRUE@am__append_6 = -Wno-pointer-sign
 @IS_GCC4_FALSE@@IS_GCC_TRUE@am__append_7 = -I. -I- -I/usr/local/include
 
--- a/configure
+++ b/configure
@@ -534,7 +534,7 @@
 #
 # Initializations.
 #
-ac_default_prefix=/usr/local
+ac_default_prefix=/usr
 ac_clean_files=
 ac_config_libobj_dir=.
 LIBOBJS=
