# correct directory /usr/local 
Index: tmux/GNUmakefile
===================================================================
--- tmux.orig/GNUmakefile	2010-03-13 23:23:20.798717192 +0100
+++ tmux/GNUmakefile	2010-03-13 23:26:35.902223278 +0100
@@ -23,7 +23,7 @@
 
 CC?= cc
 CFLAGS+= -DBUILD="\"$(VERSION)\""
-LDFLAGS+= -L/usr/local/lib
+LDFLAGS+= -L/usr/lib
 LIBS+=
 
 # Sun CC 
@@ -51,7 +51,7 @@
 endif
 endif
 
-PREFIX?= /usr/local
+PREFIX?= /usr/
 INSTALLDIR= install -d
 INSTALLBIN= install -g bin -o root -m 555
 INSTALLMAN= install -g bin -o root -m 444
@@ -80,7 +80,7 @@
 install:	all
 		$(INSTALLDIR) $(DESTDIR)$(PREFIX)/bin
 		$(INSTALLBIN) tmux $(DESTDIR)$(PREFIX)/bin/tmux
-		$(INSTALLDIR) $(DESTDIR)$(PREFIX)/man/man1
-		$(INSTALLMAN) tmux.1 $(DESTDIR)$(PREFIX)/man/man1/tmux.1
+		$(INSTALLDIR) $(DESTDIR)$(PREFIX)share/man/man1
+		$(INSTALLMAN) tmux.1 $(DESTDIR)$(PREFIX)share/man/man1/tmux.1
 
 -include .depend
Index: tmux/Makefile
===================================================================
--- tmux.orig/Makefile	2010-03-13 23:23:20.790724595 +0100
+++ tmux/Makefile	2010-03-13 23:26:35.886216224 +0100
@@ -24,7 +24,7 @@
 
 CC?= cc
 CFLAGS+= -DBUILD="\"$(VERSION)\""
-LDFLAGS+= -L/usr/local/lib
+LDFLAGS+= -L/usr/lib
 LIBS+=
 
 .ifdef FDEBUG
@@ -47,7 +47,7 @@
 .endif
 .endif
 
-PREFIX?= /usr/local
+PREFIX?= /usr
 INSTALLDIR= install -d
 INSTALLBIN= install -g bin -o root -m 555
 INSTALLMAN= install -g bin -o root -m 444
@@ -79,5 +79,5 @@
 install:	all
 		${INSTALLDIR} ${DESTDIR}${PREFIX}/bin
 		${INSTALLBIN} tmux ${DESTDIR}${PREFIX}/bin/
-		${INSTALLDIR} ${DESTDIR}${PREFIX}/man/man1
-		${INSTALLMAN} tmux.1 ${DESTDIR}${PREFIX}/man/man1/
+		${INSTALLDIR} ${DESTDIR}${PREFIX}/share/man/man1
+		${INSTALLMAN} tmux.1 ${DESTDIR}${PREFIX}/share/man/man1/
