# using setresgid() for safely dropping utmp group membership.
Index: tmux/tmux.c
===================================================================
--- tmux.orig/tmux.c	2010-03-13 23:25:52.702716773 +0100
+++ tmux/tmux.c	2010-03-13 23:25:52.714716424 +0100
@@ -186,9 +186,11 @@
 {
 	char		base[MAXPATHLEN], *path;
 	struct stat	sb;
-	u_int		uid;
+	u_int		uid,gid;
 
 	uid = getuid();
+	gid = getgid();
+
 	xsnprintf(base, MAXPATHLEN, "%s/%s/%s-%d", _PATH_VARRUN, __progname, __progname, uid);
 
 	if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST)
@@ -204,6 +206,9 @@
 		errno = EACCES;
 		return (NULL);
 	}
+	/* drop unnecessary privileges */
+ 	if (setresgid(gid, gid, gid) != 0)
+ 		return (NULL);
 
 	xasprintf(&path, "%s/%s", base, label);
 	return (path);
