# using setresgid() for safely dropping utmp group membership.
--- a/tmux.c
+++ b/tmux.c
@@ -177,6 +177,7 @@
 	u_int		uid;
 
 	uid = getuid();
+
 	xsnprintf(base, MAXPATHLEN, "%s/%s/%s-%d", _PATH_VARRUN, __progname, __progname, uid);
 
 	if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST)
--- a/server.c
+++ b/server.c
@@ -72,6 +72,13 @@
 	size_t			size;
 	mode_t			mask;
 	int			fd, mode;
+	u_int			gid;
+
+	gid = getgid();
+
+	/* drop unnecessary privileges */
+ 	if (setresgid(gid, gid, gid) != 0)
+ 		return (NULL);
 
 	memset(&sa, 0, sizeof sa);
 	sa.sun_family = AF_UNIX;
