# Enable kFreeBSD and Hurd specific configure script
--- a/configure
+++ b/configure
@@ -5637,7 +5637,7 @@
 
 		PLATFORM=darwin
 		;;
-	*linux*)
+	*linux*|GNU|GNU/kFreeBSD)
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: linux" >&5
 $as_echo "linux" >&6; }
 		PLATFORM=linux
--- a/tmux.h
+++ b/tmux.h
@@ -396,6 +396,18 @@
 	char		argv[COMMAND_LENGTH];
 };
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 64
+#endif
+
 struct msg_identify_data {
 	char		cwd[MAXPATHLEN];
 
@@ -1114,6 +1126,10 @@
 };
 RB_HEAD(status_out_tree, status_out);
 
+#ifndef IOV_MAX
+#define IOV_MAX	1024
+#endif
+
 /* Client connection. */
 struct client {
 	struct imsgbuf	 ibuf;
