# Enable kFreeBSD and Hurd specific configure script
Index: tmux/configure
===================================================================
--- tmux.orig/configure	2010-03-13 23:25:36.158219226 +0100
+++ tmux/configure	2010-03-13 23:25:52.726716354 +0100
@@ -89,7 +89,7 @@
 EOF
 	;;
 # ------------------------------------------------------------------------------
-    Linux)
+    Linux|GNU|GNU/kFreeBSD)
 	cat <<EOF >>$CONFIG_H
 #define HAVE_ASPRINTF
 #define HAVE_BZERO
Index: tmux/tmux.h
===================================================================
--- tmux.orig/tmux.h	2010-03-13 23:25:36.158219226 +0100
+++ tmux/tmux.h	2010-03-13 23:25:52.726716354 +0100
@@ -388,6 +388,10 @@
 	char		argv[COMMAND_LENGTH];
 };
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
 struct msg_identify_data {
 	char		cwd[MAXPATHLEN];
 
@@ -1065,6 +1069,10 @@
 	time_t	msg_time;
 };
 
+#ifndef IOV_MAX
+#define IOV_MAX	1024
+#endif
+
 /* Client connection. */
 struct client {
 	struct imsgbuf	 ibuf;
