from  Pierre Habouzit proposed.

diff --git a/GNUmakefile b/GNUmakefile
index 7a8adb4..54da5a6 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -23,7 +23,7 @@ FDEBUG= 1
 
 CC?= cc
 CFLAGS+= -DBUILD="\"$(VERSION)\""
-LDFLAGS+= -L/usr/lib
+LDFLAGS+= -L/usr/lib -Wl,--as-needed
 LIBS+=
 
 # Sun CC 
diff --git a/Makefile b/Makefile
index a026e0d..348cef7 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ FDEBUG= 1
 
 CC?= cc
 CFLAGS+= -DBUILD="\"$(VERSION)\""
-LDFLAGS+= -L/usr/lib
+LDFLAGS+= -L/usr/lib -Wl,--as-needed
 LIBS+=
 
 .ifdef FDEBUG
