Include cherry-picked patches from upstream.

--- a/server-client.c
+++ b/server-client.c
@@ -694,7 +694,7 @@
 	vfprintf(ctx->cmdclient->stdout_file, fmt, ap);
 	va_end(ap);
 
-	fputc('\n', ctx->cmdclient->stderr_file);
+	fputc('\n', ctx->cmdclient->stdout_file);
 	fflush(ctx->cmdclient->stdout_file);
 }
 
--- a/server-window.c
+++ b/server-window.c
@@ -109,10 +109,8 @@
 
 	if (!(w->flags & WINDOW_BELL) || wl->flags & WINLINK_BELL)
 		return (0);
-	if (s->curw == wl)
-		return (0);
-
-	wl->flags |= WINLINK_BELL;
+	if (s->curw != wl)
+		wl->flags |= WINLINK_BELL;
 
 	action = options_get_number(&s->options, "bell-action");
 	switch (action) {
--- a/window-copy.c
+++ b/window-copy.c
@@ -171,7 +171,8 @@
 	data->searchstr = NULL;
 
 	wp->flags |= PANE_FREEZE;
-	bufferevent_disable(wp->event, EV_READ|EV_WRITE);
+	if (wp->fd != -1)
+		bufferevent_disable(wp->event, EV_READ|EV_WRITE);
 
 	data->jumptype = WINDOW_COPY_OFF;
 	data->jumpchar = '\0';
@@ -234,7 +235,8 @@
 	struct window_copy_mode_data	*data = wp->modedata;
 
 	wp->flags &= ~PANE_FREEZE;
-	bufferevent_enable(wp->event, EV_READ|EV_WRITE);
+	if (wp->fd != -1)
+		bufferevent_enable(wp->event, EV_READ|EV_WRITE);
 
 	if (data->searchstr != NULL)
 		xfree(data->searchstr);
